src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -157,6 +157,20 @@ typedef struct MetadataCacheData
 	Oid jsonbExtractPathFuncId;
 	bool databaseNameValid;
 	char databaseName[NAMEDATALEN];
+
+	/* cached lookups for the tdigest extension */
+	Oid tdigestExtensionSchema;
```

Comment:
So there's caching issues, & HLL/topn isn't in here, what's the cost of not caching this?

Reply:
The cost is that it will need to perform up to 10 lookups into the catalog (2x once for the extension schema and once for the function/type oid). To come to a conclusion which aggregate we are targeting.
Secondly each worker and coordinator part of the query requires ~3 lookups each to construct the query part.

coming to up to 26 catalog lookups.
Luckily this only happens if the aggregate starts with `tdigest`, so it will only perform them if there is a high likely hood we are actually querying for a tdigest.

Given the execution time is already high on queries where you want to use tdigest these lookups during planning probably are not significant to the execution of the query. Hence Marco already suggesting to not cache them.

I am moving these functions to their own file to not pollute the metadata cache file more.